shell/platform/android/io/flutter/embedding/FlutterActivity.java

```diff
@@ -0,0 +1,236 @@
+  protected void onNewIntent(Intent intent) {
+    boolean isRunAction = Intent.ACTION_RUN.equals(intent.getAction());
+    if (isDebuggable() && isRunAction) {
+      // We're in debug mode, and processing an ACTION_RUN, so we will hot reload Flutter.
+      String route = intent.getStringExtra("route");
+      String appBundlePath = intent.getDataString();
+      if (appBundlePath == null) {
+        // Fall back to the installation path if no bundle path was specified.
+        appBundlePath = FlutterMain.findAppBundlePath(getApplicationContext());
+      }
+      flutterFragment.hotReload(route, appBundlePath);
+    } else {
+      // We're not in debug mode, or we're processing an action other than ACTION_RUN. Therefore,
+      // we will not execute a hot reload on Flutter.  But, we will forward this event to the
+      // Fragment so that the Intent can be delivered to any interested plugins.
+      flutterFragment.onNewIntent(intent);
+    }
+  }
+
+  @Override
+  public void onUserLeaveHint() {
+    flutterFragment.onUserLeaveHint();
+  }
+
+  /**
+   * Sets up a {@code FrameLayout} that takes up all available space in the {@code Activity}. This
+   * {@code FrameLayout} will hold a {@code FlutterFragment}, which displays a {@code FlutterView}.
+   */
+  private void createFlutterFragmentContainer() {
+    FrameLayout container = new FrameLayout(this);
+    container.setId(CONTAINER_ID);
+    container.setLayoutParams(new ViewGroup.LayoutParams(
+        ViewGroup.LayoutParams.MATCH_PARENT,
+        ViewGroup.LayoutParams.MATCH_PARENT
+    ));
+    setContentView(container);
+  }
+
+  /**
+   * If no {@code FlutterFragment} exists in this {@code FlutterActivity}, then a {@code FlutterFragment}
+   * is created and added. If a {@code FlutterFragment} does exist in this {@code FlutterActivity}, then
+   * a reference to that {@code FlutterFragment} is retained in {@code flutterFragment}.
+   */
+  private void createFlutterFragment() {
+    FragmentManager fragmentManager = getFragmentManager();
+
+    flutterFragment = (FlutterFragment) fragmentManager.findFragmentByTag(TAG_FLUTTER_FRAGMENT);
+
+    if (flutterFragment == null) {
+      // No FlutterFragment exists yet. This must be the initial Activity creation. We will create
+      // and add a new FlutterFragment to this Activity.
+      flutterFragment = FlutterFragment.newInstance(
+          isSplashScreenDesired(),
+          getInitialRoute(),
+          getAppBundlePath()
+      );
+      fragmentManager
+          .beginTransaction()
+          .add(CONTAINER_ID, flutterFragment, TAG_FLUTTER_FRAGMENT)
+          .commit();
+    }
+  }
+
+  /**
+   * Sets the color of this {@code Activity}'s status bar to a translucent black, i.e., a dark
+   * overlay.
+   *
+   * Beginning with Android Lollipop, the status bar at the top of the screen can have its color
+   * set at runtime.
+   */
+  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+  private void configureStatusBarColor() {
```

Comment:
Thanks for refactoring these into methods

Reply:
👍